### PR TITLE
enable incremental backups for sharepoint libraries

### PR DESCRIPTION
**Pull request by ryanfkeepers**
introduces incremental backups for sharepoint libraries.

---

#### Does this PR need a docs update or release note?

- [x] :white_check_mark: Yes, it's included

#### Type of change

- [x] :sunflower: Feature

#### Issue(s)

* #3136

#### Test Plan

- [x] :muscle: Manual
- [x] :zap: Unit test
- [x] :green_heart: E2E

### CODE REVIEW

**Review by meain — APPROVED** [review]
LGTM, few small things.

File: src/internal/common/crash/crash.go
Comment (ashmrtn) [inline src/internal/common/crash/crash.go:1]: please move changes that are not directly related to the PR in question to a different PR

File: src/internal/kopia/merge_details.go [hunk 1]
```diff
@@ -4,13 +4,18 @@ import (
...
 	"github.com/alcionai/corso/src/pkg/path"
 )
 
 type DetailsMergeInfoer interface {
 	// ItemsToMerge returns the number of items that need to be merged.
 	ItemsToMerge() int
+	// TODO(keepers): remove once sharepoint fully supports metadata files.
+	// ItemsToMerge returns the number of items that need to be merged,
+	// excluding any item that ends in .meta or .dirmeta
+	ItemsToMergeSansMeta() int
```
Comment (ashmrtn) [inline src/internal/kopia/merge_details.go:18]: can you expand on the need for this? We shouldn't be adding metadata files to backup details at all by this point and we shouldn't be pulling them forward from base snapshots

Comment (ashmrtn) [reply src/internal/kopia/merge_details.go:18]: SP should also already support metadata files since those are now where file names are stored. It just doesn't support permissions right now

File: src/internal/connector/data_collections.go [hunk 2]
```diff
@@ -64,11 +65,16 @@ func (gc *GraphConnector) ProduceBackupCollections(
...
+	)
+
 	switch sels.Service {
 	case selectors.ServiceExchange:
-		colls, excludes, err := exchange.DataCollections(
+		colls, excludes, err = exchange.DataCollections(
 			ctx,
-			sels,
+			owner,
 			sels,
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/connector/data_collections.go:78]: done

File: src/internal/operations/backup_test.go [hunk 3]
```diff
@@ -212,6 +212,10 @@ func (m *mockDetailsMergeInfoer) ItemsToMerge() int {
 	return len(m.repoRefs)
 }
 
+func (m *mockDetailsMergeInfoer) ItemsToMergeSansMeta() int {
+	return -1
```
[the remark this answers is not in the document]
Comment (ryanfkeepers) [reply src/internal/operations/backup_test.go:216]: We'd have to pass in a `*testing.T`.  I assume the -1 will catch most cases.  Can be revisited if/when wanted.

File: src/internal/common/crash/crash.go
Comment (ryanfkeepers) [reply src/internal/common/crash/crash.go:1]: @ashmrtn see: https://github.com/alcionai/corso/pull/3201 as a pre-PR to this one.

**Review by ashmrtn — APPROVED** [review]
